src/runtime_src/xdp/profile/database/static_info/device_info.cpp

```diff
@@ -26,273 +26,369 @@ namespace xdp {
 
   DeviceInfo::~DeviceInfo()
   {
-    for (auto i : loadedXclbins) {
-      delete i ;
+    for (auto cfg : loadedConfigInfos) {
+      delete cfg;
     }
-    loadedXclbins.clear() ;
+    loadedConfigInfos.clear() ;
   }
 
   std::string DeviceInfo::getUniqueDeviceName() const
   {
     return deviceName + "-" + std::to_string(deviceId) ;
   }
 
+  //TODO: Check the implementation here, return unified UUID.
   xrt_core::uuid DeviceInfo::currentXclbinUUID()
   {
-    if (loadedXclbins.size() <= 0)
+    if (loadedConfigInfos.size() <= 0)
       return xrt_core::uuid() ;
-    return loadedXclbins.back()->uuid ;
+    return currentConfig()->currentXclbins.back()->uuid ;
```

Comment:
change to concatenate the all xclbins uuid to form an unique UUID.

To check if  we cross refer to what we have stored on xrt_core device